duckduckgo/iOS
# Feature flagging with autofill subfeatures

Task/Issue URL: https://app.asana.com/0/72649045549333/1203676877763836
Tech Design URL: https://app.asana.com/0/481882893211075/1203938569856702
https://app.asana.com/0/481882893211075/1203984214904257
CC:

Description:

This PR and its corresponding [BSK PR](https://github.com/duckduckgo/BrowserServicesKit/pull/270):
- adds subfeatures support to the Privacy Config component
- moves the FeatureFlagger from iOS into BSK and updates it for the requirements gathered during this project
- moves the InternalUserDecider from iOS into BSK

To understand why I ended up at the solution I did, it's worth pointing out some requirements:
- We should be able to remotely toggle features using the privacy config
- Remote toggling shouldn't necessarily be possible while the feature is in development, only when we're close to release. Up until this point, we should stick with having them enabled internally alone by authenticating as an internal user.
- The current state of a given feature (particularly during development) may well be different in iOS vs macOS and we need a way to express that.
- Some features (autofill included) may be developed (and in the future disabled) only partially, introducing the need for nested features ("subfeatures") in the Privacy Configuration.

Additionally, this stops in-development features being enabled by default in dev builds as discussed here: https://app.asana.com/0/1200194497630846/1203752002572023/f

**Steps to test this PR**:
1. Check that any features configured using the privacy-config still work as expected (particularly autofill).
2. Check feature flags (autofill, debug menu, sync)



**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
